opm/simulators/wells/StandardWellConnections.cpp

```diff
@@ -364,94 +364,49 @@ computePropertiesForPressures(const WellState& well_state,
-                    if (gasrate > 0) {
-                        rvw = waterrate / gasrate;
-                    }
-                    rvw = std::min(rvw, rvwmax_perf[perf]);
-                }
-                if (rv > 0.0 || rvw > 0.0){
-                    b_perf[gaspos] = FluidSystem::gasPvt().inverseFormationVolumeFactor(region_idx, temperature, p_avg, rv, rvw);
-                }
-                else {
-                    b_perf[gaspos] = FluidSystem::gasPvt().saturatedInverseFormationVolumeFactor(region_idx, temperature, p_avg);
-                }
-            } else if (oilPresent) {
-                //no water
+            double rvw = 0.0;
+            double rv = 0.0;
+            if (oilPresent) {
                 const double oilrate = std::abs(ws.surface_rates[pu.phase_pos[Oil]]); //in order to handle negative rates in producers
                 rvmax_perf[perf] = FluidSystem::gasPvt().saturatedOilVaporizationFactor(region_idx, temperature, p_avg);
                 if (oilrate > 0) {
                     const double gasrate = std::abs(ws.surface_rates[pu.phase_pos[Gas]]) - (Indices::enableSolvent ? ws.sum_solvent_rates() : 0.0);
-                    double rv = 0.0;
                     if (gasrate > 0) {
                         rv = oilrate / gasrate;
                     }
                     rv = std::min(rv, rvmax_perf[perf]);
-
-                    b_perf[gaspos] = FluidSystem::gasPvt().inverseFormationVolumeFactor(region_idx, temperature, p_avg, rv, 0.0 /*Rvw*/);
-                }
-                else {
-                    b_perf[gaspos] = FluidSystem::gasPvt().saturatedInverseFormationVolumeFactor(region_idx, temperature, p_avg);
                 }
-            } else if (waterPresent) {
-                //no oil
+            }
+            if (waterPresent) {
                 const double waterrate = std::abs(ws.surface_rates[pu.phase_pos[Water]]); //in order to handle negative rates in producers
                 rvwmax_perf[perf] = FluidSystem::gasPvt().saturatedWaterVaporizationFactor(region_idx, temperature, p_avg);
                 if (waterrate > 0) {
                     const double gasrate = std::abs(ws.surface_rates[pu.phase_pos[Gas]]) - (Indices::enableSolvent ? ws.sum_solvent_rates() : 0.0);
-                    double rvw = 0.0;
                     if (gasrate > 0) {
                         rvw = waterrate / gasrate;
                     }
                     rvw = std::min(rvw, rvwmax_perf[perf]);
-
-                    b_perf[gaspos] = FluidSystem::gasPvt().inverseFormationVolumeFactor(region_idx, temperature, p_avg, 0.0 /*Rv*/, rvw);
                 }
-                else {
-                    b_perf[gaspos] = FluidSystem::gasPvt().saturatedInverseFormationVolumeFactor(region_idx, temperature, p_avg);
-                }
-
-            } else {
-                b_perf[gaspos] = FluidSystem::gasPvt().saturatedInverseFormationVolumeFactor(region_idx, temperature, p_avg);
```

Comment:
Will we need this line anymore? I think it will only called when only `gas` exists?

Reply:
or you think `rv==0.` and `rvw==0.0`, we only need `FluidSystem::gasPvt().inverseFormationVolumeFactor(region_idx, temperature, p_avg, rv, rvw);`?